modules/02_01_Guides_Choosing_compiler_options.md

```diff
@@ -0,0 +1,175 @@
+# Choosing module-related compiler options
+
+## I’m writing an app
+
+A single tsconfig.json can only represent a single environment, both in terms of what globals are available and in terms of how modules behave. If your app contains server code, DOM code, web worker code, test code, and code to be shared by all of those, each of those should have its own tsconfig.json, connected with [project references](https://www.typescriptlang.org/docs/handbook/project-references.html#handbook-content). Then, use this guide once for each tsconfig.json. For library-like projects within an app, especially ones that need to run in multiple runtime environments, use the “[I’m writing a library](#im-writing-a-library)” section.
+
+### I’m using a bundler
+
+In addition to adopting the following settings, it’s also recommended _not_ to set `{ "type": "module" }` or use `.mts` files in bundler projects for now. [Some bundlers](https://andrewbranch.github.io/interop-test/#synthesizing-default-exports-for-cjs-modules) adopt different ESM/CJS interop behavior under these circumstances, which TypeScript cannot currently analyze with `"moduleResolution": "bundler"`. See [issue #54102](https://github.com/microsoft/TypeScript/issues/54102) for more information.
+
+```json5
+{
+  "compilerOptions": {
+    // This is not a complete template; it only
+    // shows relevant module-related settings.
+    // Be sure to set other important options
+    // like `target`, `lib`, and `strict`.
+
+    // Required
+    "module": "esnext",
+    "moduleResolution": "bundler",
+    "esModuleInterop": true,
+
+    // Consult your bundler’s documentation
+    "customConditions": ["module"],
+
+    // Recommended
+    "noEmit": true, // or `emitDeclarationOnly`
+    "allowImportingTsExtensions": true,
+    "allowArbitraryExtensions": true,
+    "verbatimModuleSyntax": true, // or `isolatedModules`
```

Comment:
I would consider IM "soft deprecated" in favor of VMS, so maybe worth downplaying